Lib/asyncio/base_events.py

```diff
@@ -1993,23 +1997,24 @@ def _run_once(self):
 
         # Handle 'later' callbacks that are ready.
         end_time = self.time() + self._clock_resolution
+        ready = self._ready
```

Comment:
While I understand the reason of this micro optimization, I'm not sure it's worth the change. If you want an even better optimized call, you should use `mark_ready = self._ready.append` (and same for the popleft). Similarly, `self._scheduled` would be more worth being cached but you only do it for `self._ready` (there are more calls to `self._scheduled` than to `self._ready` in the first loop).

Reply:
I made some adjustments in fdda41627c72209fe1ed5c4db16489da427d0eb2

I think `ready.append` is more of a rare case as much iterations will break right away so I left that as-is

Reply:
![run_once](https://github.com/user-attachments/assets/cf9c4fd0-a6f9-41d3-ba27-8dee035cc1a2)

`run_once` is probably worth optimizing some more since its the most expensive part that isn't already implemented in native code but I didn't want to increase the scope of this PR.

Reply:
Actually, in the latest Python versions, the cost of accessing an attribute becomes almost negligible. My comment was more about: "I don't think you should cache *any* of the attribute as a local variable". I think the bottleneck is more about the actual call and not the time it takes to retrieve the attribute and call it (I don't know if I'm clear enough).